horace_core/GUI/horace.m

```diff
@@ -49,6 +49,24 @@
     strcmp(element_class,'d3d') || strcmp(element_class,'d4d') ||...
     strcmp(element_class,'sqw');
 
+function [vec_val,ok] = read_vector(str_val)
+% read vector value from a string, obtained from GUI
+ok = true;
+s1=strfind(str_val,'['); s2=strfind(str_val,']');
+if isempty(s1) && isempty(s2)
+    sv_new=textscan(str_val,'%f','delimiter',',');
+elseif ~isempty(s1) && ~isempty(s2)
+    str_val=str_val(s1+1:s2-1);
```

Comment:
This function just ignores junk which could be bad. A valid input is
`ghosts[1,2,3]inthecode`
I would recommend you use `startsWith` and `endsWith`  and `(2:end-1)` for more strict input validation.

Reply:
despite these operations are introduced in 2016b, I am still a bit reluctant to use operations, not compatible with the previous versions. May be should start changing this.

Reply:
Given we support 2018b+ I don't think we should be avoiding them. At the very least check `1` and `end` only.

Reply:
Had some thinking about the changes. `startWith` and `endWith` look much clearer, but substantially change the code behavior.  Will go well together with exceptions handling in the `read_vector` but I do not have desire and time to do this improvement now.